core-services/importExport/src/main/java/org/silverpeas/core/importexport/attachment/AttachmentImportExport.java

```diff
@@ -71,8 +72,10 @@ public AttachmentImportExport(final UserDetail user) {
     this.user = user;
   }
 
-  public List<AttachmentDetail> importAttachments(String pubId, String componentId,
-      List<AttachmentDetail> attachments, boolean indexIt) throws FileNotFoundException {
+  public List<Pair<AttachmentDetail, SimpleDocument>> importAttachments(String pubId,
+      String componentId, List<AttachmentDetail> attachments, boolean indexIt) {
```

Comment:
Why return a pair of `AttachmentDetail` and its `SimpleDocument` container instead of simply returning the `SimpleDocument`. We can get the `AttachmentDetail` from the `SimpleDocument` instance.
Yes, the embedded `AttachementDetail` in the `SimpleDocument` is a copy of the one in the list passed as argument of the function (and hence a copy of the one in the pair) but it is this one that is actually saved and it has all the required information about the attachment itself. This will simplify the method signature and avoid any interrogations of the readers about such a signature (and possible incoherence between the `AttachmentDetail` instance in the pair with the one wrapped by the `SimpleDocument` instance in the pair)

Reply:
The `AttachmentDetail` you are talking about, the one provided by SimpleDocument, is not the one returned by this signature. I mean, it's not the same implementation. The `AttachmentDetail` returned is a bean implemented in the `importexport` package and is used to handle special things that only that package needs.
You can imagine that if I could have done things differently, with a simpler signature return, I would have.

Reply:
ok, I was misled by the name of the class